source/common/grpc/async_client_impl.h

```diff
@@ -49,9 +49,10 @@ class AsyncClientImpl final : public AsyncClient<RequestType, ResponseType> {
 
   AsyncStream<RequestType>* start(const Protobuf::MethodDescriptor& service_method,
                                   AsyncStreamCallbacks<ResponseType>& callbacks) override {
+    const Optional<std::chrono::milliseconds> no_timeout;
```

Comment:
I checked and it does get copied inside the underlying `Http::AsyncClient::Stream`, in the owned `RouteEntryImpl` https://github.com/lyft/envoy/blob/0efa18c36d6f789562690b5053c2c4b00987979e/source/common/http/async_client_impl.h#L123. 

More generally, this got me wondering about whether we should have a better convention for handling reference ownership semantics in Envoy - it's hard to tell from the type signature of a method alone whether it will store a reference for later use or copy. I think there's a fair bit of mixed use of this in Envoy.